### ISSUE TITLE
[ARAB] Reported Fails

### ISSUE THREAD

**Issue by vv-monsalve** [TypeTogether/Playpen-Sans#23]
Hi @josescaglione, @casasin 

I'll start collecting in this issue the Fail or warns reported related to the Arabic script

<details>
    <summary>🔥 <b>FAIL</b> Check that Arabic spacing symbols U+FBB2–FBC1 aren't classified as marks. <a href="https://fontbakery.readthedocs.io/en/stable/fontbakery/checks/universal.arabic.html#"></a></summary>
    <div>

> Unicode has a few spacing symbols representing Arabic dots and other marks,
> but they are purposefully not classified as marks.
> 
> Many fonts mistakenly classify them as marks, making them unsuitable
> for their original purpose as stand-alone symbols to used in pedagogical
> contexts discussing Arabic consonantal marks.
> 

> Original proposal: https://github.com/googlefonts/fontbakery/issues/4295

* 🔥 **FAIL** <p>&quot;tahabove-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

* 🔥 **FAIL** <p>&quot;dotabove-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

* 🔥 **FAIL** <p>&quot;dotbelow-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

* 🔥 **FAIL** <p>&quot;twodotshorizontalabove-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

* 🔥 **FAIL** <p>&quot;twodotshorizontalbelow-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

* 🔥 **FAIL** <p>&quot;threedotsupabove-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

* 🔥 **FAIL** <p>&quot;threedotsdownbelow-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

* 🔥 **FAIL** <p>&quot;wasla-ar&quot; is defined in GDEF as a mark (class 3).</p>
 [code: mark-in-gdef]

</div>

**Comment by yanone** [comment]
I’m not aware of `tahabove-ar` or `wasla-ar` ever used as standalone characters in modern times (tho I could be mistaken); only ever as components in letters such as `ٹ`, much less so for the dot components. 

These have assigned unicodes since the ancient times but I’ve never seen them used and I’m not sure why they even exist in modern fonts.
Therefore I think the checks can be ignored regardless of how these glyphs are defined in GDEF.

To be sure, I’ve compared the shaping to Noto Naskh and found the `wasla-ar` to break the shaping in both fonts while the `tah-ar` works in both except that in Noto it’s not getting attached properly (sitting too high).

In conclusion, the fonts are fine. Ignore the checks.

**Comment by yanone** [comment]
![Bildschirmfoto 2024-10-18 um 14 43 07](https://github.com/user-attachments/assets/367a3b48-e63c-448b-85ba-e81330471105)

**Comment by vv-monsalve** [comment]
Thank you, @yanone, for reviewing this. Please confirm if, following this, the #27 can be ignored aswell.

**Comment by vv-monsalve** [comment]
Hi, @josescaglione. Please check the first two cases of these shaping Fail reported for the Arabic font.

🔥 <b>FAIL</b> Shapes languages in all GF glyphsets. <a href="https://fontbakery.readthedocs.io/en/stable/fontbakery/checks/vendorspecific.googlefonts.glyphset.html#"></a>
    <div>
> This check uses a heuristic to determine which GF glyphsets a font supports.
> Then it checks the font for correct shaping behaviour for all languages in
> those glyphsets.

> Original proposal: https://github.com/googlefonts/fontbakery/issues/4147

* 🔥 **FAIL** <p>GF_TransLatin_Pinyin glyphset:</p>
<table>
<thead>
<tr>
<th align="left">FAIL messages</th>
<th align="left">Languages</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">.medi version of ARABIC LETTER NOON; both buffers returned gid2=1+0</td>
<td align="left">gid1550=1@11,-126+0</td>
</tr>
<tr>
<td align="left">.init version of ARABIC LETTER NOON; both buffers returned gid2=0+0</td>
<td align="left">gid1550=0@11,-126+0</td>
</tr>
<tr>
<td align="left">in Farsi, &quot;saria heh&quot; should use a small yeh above, not hamza; both buffers returned gid1563=0@-60,54+0</td>
<td align="left">gid1349=0+505</td>
</tr>
</tbody>
</table>

**Comment by simoncozens** [comment]
Please ignore the "saria heh" fail. However, the NOON fail is important. NOON is not forming medi or init forms. Compare, for the text ببب تتت ننن:
![shape](https://github.com/user-attachments/assets/913e0dfc-75af-425b-b60c-5e0d710c1a9e)

I see that there are init/medi/fina rules for noon in `features/ara-init-medi-fina.fea`. However,  noon is already decomposed to `noonghunna-ar dotabove-ar` in `ccmp`, before these rules fire. So the rules in init/medi/fina need to apply to `noonghunna-ar` instead of `noon-ar`.

**Comment by casasin** [comment]
Hi, this has been fixed in Commit [2e62a99](https://github.com/TypeTogether/Playpen-Sans/commit/2e62a9906c307872eff19771d05d1622a4a7f564).
Please, confirm and if so, close.

**Comment by vv-monsalve** [comment]
The Fail is still reported, but the Noon is working now. 

<img width="536" alt="Screenshot 2024-12-11 at 11 26 45" src="https://github.com/user-attachments/assets/c3d8413b-fc75-454a-ad84-ad7a717d14e1" />
